setup.py

```diff
@@ -39,7 +39,8 @@
     # we are rather picky about msgpack versions, because a good working msgpack is
     # very important for borg, see https://github.com/borgbackup/borg/issues/3753
     # best versions seem to be 0.4.6, 0.4.7, 0.4.8 and 0.5.6:
-    'msgpack-python >=0.4.6, <=0.5.6, !=0.5.0, !=0.5.1, !=0.5.2, !=0.5.3, !=0.5.4, !=0.5.5',
+    'msgpack-python <0.5;python_version=="3.4"',
+    'msgpack >=0.5.6;python_version >="3.5"',
```

Comment:
the problem is just that this is not what we want:
- we do not want to accept almost any early msgpack version on 3.4
- we do not want to require a very recent on >=3.5

Reply:
Also: i tested 0.5.6 on py34, it was ok (see that big msgpack issue on our issue tracker). So it can be used on py34.

Reply:
Also: msgpack made a big mess in 0.5.x (see same issue), so I won't trust in future releases "just working" without checking first - so no way for ">=0.5.6".

Reply:
Also note how on github only 0.4.6 is marked as "release" and everything newer is "pre-release"...

Reply:
> we do not want to require a very recent on >=3.5

Why not? Having to deal with two different naming schemes on >=3.5 (msgpack-python is not guaranteed to exist, it emphatically does not exist on Arch Linux) does not seem very reasonable. Moreover, having to fight with pip to ensure that the dummy package name actually installs the thing it's supposed to install, is just asking for trouble!

Meanwhile, anyone who is *updating* to new versions of borg, should have no problem *updating* msgpack too. I'm not sure I understand the motivation to support old stuff when there's compelling reasons to not do so, and *zero* motivation to do so.

This would work flawlessly for people using pip. By default they get the latest version of msgpack and the latest version of borg.

If this is solely about distro support for old versions of msgpack, then any distro which does not want to upgrade msgpack can be assumed to not want to upgrade borg either... holding the project hostage to old distros and requiring the *newer* distros to be the ones to patch borg in order to remain functional, seems entirely backwards.

> So it can be used on py34.

I did not see that anywhere. BTW are you planning on continually testing newer msgpack releases on python 3.4 before adding it for python>=3.5 at all? If it stops working on python 3.4 will you just freeze the pinned msgpack version for every python release?

> Also: msgpack made a big mess in 0.5.x (see same issue), so I won't trust in future releases "just working" without checking first - so no way for ">=0.5.6".

I don't see why "it's slow, but only if you build from git without cython installed" or "it prints warnings which aren't wrong", or "it removed private APIs which are not permitted to be used, and broke bad code as a result", are reasons to fundamentally distrust it. The biggest issue I see is that the dummy package causes issues for people who do not migrate to the new name...

Plus one regression fixed later the same day.

And I'd guess this is why Pipfile.lock exists anyway.

> Also note how on github only 0.4.6 is marked as "release" and everything newer is "pre-release"...

I also note how *you're* the one who opened msgpack/msgpack-python#298 and discovered that these tags have no meaning. I mean, I guess I don't know why they're marked as anything at all, maybe AppVeyor does this?

It's a bit disingenuous to then bring this as a naysaying argument here...